### PR TITLE
OPTForCasualLM Support (facebook/opt Series)

### PR DESCRIPTION
**Pull request by b8zhong**
## Motivation

Expand support for the [OPT series](https://huggingface.co/facebook/opt-125m), this includes from `opt-125m` to `opt-175b`.

> "We present Open Pretrained Transformers (OPT), a suite of decoder-only pre-trained transformers ranging from 125M to 175B parameters, which we aim to fully and responsibly share with interested researchers. We train the OPT models to roughly match the performance and sizes of the GPT-3 class of models, while also applying the latest best practices in data collection and efficient training."

## Modifications

Add the required modelling code. Note that since the activation is ReLU I added it to the activation registry for it to work.

## Evaluation (MMLU):

`python3 bench_other.py --nsub 10 --backend vllm` and `python3 bench_sglang.py --nsub 10` on `facebook/opt-125m`:

| Subject                   | # Questions | Accuracy (SGLang) | Accuracy (vLLM) | Absolute Difference |
|---------------------------|-------------|--------------------|------------------|----------------------|
| abstract_algebra          | 100         | 0.280              | 0.270            | 0.010                |
| anatomy                   | 135         | 0.237              | 0.237            | 0.000                |
| astronomy                 | 152         | 0.263              | 0.263            | 0.000                |
| business_ethics           | 100         | 0.010              | 0.010            | 0.000                |
| clinical_knowledge        | 265         | 0.211              | 0.208            | 0.003                |
| college_biology           | 144         | 0.104              | 0.104            | 0.000                |
| college_chemistry         | 100         | 0.230              | 0.230            | 0.000                |
| college_computer_science  | 100         | 0.330              | 0.330            | 0.000                |
| college_mathematics       | 100         | 0.260              | 0.260            | 0.000                |
| college_medicine          | 173         | 0.150              | 0.156            | 0.006                |
| **Average**               | —           | **0.205**          | **0.204**        | **0.001**            |

### CODE REVIEW

File: python/sglang/srt/layers/activation.py [hunk 1]
```diff
@@ -157,6 +157,7 @@ def weight_loader(self, param: nn.Parameter, loaded_weight: torch.Tensor):
 _ACTIVATION_REGISTRY = {
     "gelu": nn.GELU(),
     "gelu_pytorch_tanh": nn.GELU(approximate="tanh"),
+    "relu": nn.ReLU(),
```
[the remark this answers is not in the document]
Comment (b8zhong) [reply python/sglang/srt/layers/activation.py:160]: Do not add useless comments 🙃

File: python/sglang/srt/models/opt.py [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+            embed_dim,
+            bias=bias,
+            quant_config=quant_config,
+            prefix=add_prefix("out_proj", prefix),
+        )
+        self.attn = RadixAttention(
+            self.num_heads,
+            self.head_dim,
+            self.scaling,
+            num_kv_heads=self.num_heads,  # OPT uses same number of heads for K/V
```
[the remark this answers is not in the document]
Comment (b8zhong) [reply python/sglang/srt/models/opt.py:98]: This is within the OPT modelling file...

**Comment by b8zhong** [comment]
https://github.com/sgl-project/sglang/pull/10165
